src/material.cpp

```diff
@@ -469,13 +480,125 @@ void pyne::Material::write_hdf5(std::string filename, std::string datapath,
   H5Sclose(metadatapace);
   H5Tclose(attrtype);
 
+  delete[] mat_data;
+
+}
+
+
+void pyne::Material::write_hdf5(std::string filename, std::string datapath,
+                                float row, int chunksize) {
+  // Turn off annoying HDF5 errors
+  H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+
+  // Set file access properties so it closes cleanly
+  hid_t fapl;
+  fapl = H5Pcreate(H5P_FILE_ACCESS);
+  H5Pset_fclose_degree(fapl, H5F_CLOSE_STRONG);
+  // Create new/open datafile.
+  hid_t db;
+  if (pyne::file_exists(filename)) {
+    bool ish5 = H5Fis_hdf5(filename.c_str());
+    if (!ish5) throw h5wrap::FileNotHDF5(filename);
+    db = H5Fopen(filename.c_str(), H5F_ACC_RDWR, fapl);
+
+    // Check is /material type
+    herr_t status;
+    H5O_info_t object_info;
+    status = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+    status = H5Oget_info_by_name(db, "/material", &object_info, H5P_DEFAULT);
+
+    if (object_info.type == H5O_TYPE_DATASET)
+       {
+        // Check if path exists
+        bool datapath_exists = h5wrap::path_exists(db, datapath);
+        // no Group but path fallback on old method using default
+        if (datapath_exists) {
+          H5Fclose(db);
+          write_hdf5(filename, datapath, "/nucid", row, chunksize);
+          return;
+        }
+      } else if ( object_info.type >= H5O_TYPE_NTYPES || object_info.type < H5O_TYPE_UNKNOWN ) {
```

Comment:
are these the conditions that define that a group does not already exist?  Is there no simpler way?

Reply:
This may need more comments for which case is which in the if statements